src/App.tsx

```diff
@@ -1,61 +1,109 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './Types';
+import { TodoList } from './components/TodoList';
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => todo.userId === user.id) || null,
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+
+  const [titleError, setTitleError] = useState('');
+  const [selectUserError, setSelectUserError] = useState('');
+  const [allTodos, setAllTodos] = useState<Todo[]>(todos);
+
+  function reset() {
+    setTitleError('');
+    setSelectUserError('');
+    setTitle('');
+    setSelectUser(0);
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError('enter a title');
+    }
+
+    if (!selectUser) {
+      setSelectUserError('choose a user');
+    }
+
+    if (title && selectUser) {
+      const biggestId: number = allTodos.reduce((a, c) =>
+        a.id > c.id ? a : c,
+      ).id;
+
+      const newTodo: Todo = {
+        id: biggestId + 1,
+        title,
+        completed: false,
+        userId: +selectUser,
+        user: usersFromServer.find(user => +selectUser === user.id) || null,
```

Comment:
Create separate function for `usersFromServer.find(user => +selectUser === user.id) || null,` because you use it twice(here and above)